src/drivers/net/dwc_gmac/dwc_gmac.c

```diff
@@ -395,17 +401,17 @@ static uint32_t dwc_setup_rxdesc(struct dwc_priv *priv, int idx) {
 
 	assert(idx < RX_DESC_QUANTITY);
 
-	desc = &priv->rxdesc_ring_paddr[idx];
+	desc = &priv->rxdesc_ring[idx];
 
 	memset(desc, 0, sizeof(*desc));
 	data_mem_barrier();
 
-	desc->basic.des2 = (uint32_t)&rx_buffers[idx][0];
+	desc->basic.des2 = (uint32_t)dma_map_single(NULL, &rx_buffers[idx][0], ETH_FRAME_LEN, DMA_FROM_DEVICE);
```

Comment:
According to Linux documentation, `dma_map_single(..., DMA_FROM_DEVICE)` must be followed by one of `dma_sync_*()` calls before accessing this memory by the software; this `dma_sync_*()` functions are not trivial for MIPS architecture, for example.

```
	DMA_FROM_DEVICE synchronisation must be done before the driver
	accesses data that may be changed by the device.  This memory should
	be treated as read-only by the driver.  If the driver needs to write
	to it at any point, it should be DMA_BIDIRECTIONAL (see below).
```

```
	void
	dma_sync_single_for_cpu(struct device *dev, dma_addr_t dma_handle,
				size_t size,
				enum dma_data_direction direction)
```

Symmetric issue with `dma_map_single(..., DMA_TO_DEVICE)` / `dma_sync_single_for_device()` for tx buffers.